src/main/java/com/lafaspot/icap/client/IcapClient.java

```diff
@@ -137,6 +137,21 @@ public IcapClient(final int threads, final int connectTimeout, final int inactiv
     }
 
     /**
+     * API to scan a file (without session pool), will return a future object to be polled for result.
+     *
+     * @param server URI pointing to the Symantec AV scan server
+     * @param filename name of the file to be scanned
+     * @param toScanFile byte stream of the file to be scanned
+     * @return the future object
+     * @throws IcapException on failure
+     */
+    public Future<IcapResult> scanFileWithoutSessionReuse(@Nonnull final URI server, @Nonnull final String filename, @Nonnull final byte[] toScanFile)
```

Comment:
Can you check for maxAllowedSession count to be 0, if not throw an ICAPException